exonum-java-binding/doc/Migration_guide_0.6.md

```diff
@@ -0,0 +1,165 @@
+# Migrating Java Service to 0.6
+
+## Update the Service
+
+### Service Module Requirements
+
+Modify the service module so that it:
+1. Is `public`
+1. Extends [`AbstractServiceModule`][abstract-service-module-javadoc]
+1. Has `@org.pf4j.Extension` annotation.
+
+See the [dependencies management section][dep-management-docs] for details.
+
+[abstract-service-module-javadoc]: https://todo.com
+<!-- todo: Dependencies Management -->
+[dep-management-docs]: https://todo.com
+
+### Service Packaging Requirements
+ 
+Exonum Java 0.6.0 brings a new packaging format of services. Service
+classes and resources must be packaged in a single JAR archive
+with some metadata set.
+
+To update an existing service:
+
+1. Configure a maven-assembly-plugin or maven-shade-plugin to create
+   a single JAR with all service dependencies.
+   See [How to Build Service Artifact][how-to-build-docs] section for details.
+1. Add the following manifest entries to the JAR:
+    - "Plugin-Id" equal to "${project.groupId}:${project.artifactId}:${project.version}"
+    - "Plugin-Version" equal to "${project.version}"
+1. Add a "provided"-scoped dependency on PF4J.
```

Comment:
I wonder if the annotation processor present in PF4J will kick in if it remains a transitive dependency of Exonum? In other words, do users **have to** specify it explicitly?